src/test/resources/UserRegistration.feature

```diff
@@ -0,0 +1,47 @@
+Feature: User Registration
+  As a user (/admin)
+  I want to be able to register (users) with username and password
+
+  Background:
+    Given database is empty
+    And user with id 1234 exists and has username "user", password "secure_password" and refreshToken "token"
+
+  Scenario: Successful registration with username, password and password confirmation.
+    When user requests registration with username "kangaroo", password "pig-system" and password confirmation "pig-system"
+    Then response status code is 201
+    And response contains refreshToken "new-token" and the user with id "1235"
+
+  Scenario: Successful registration with username, password and password confirmation; password matches password of other users.
+    When user requests registration with username "kangaroo", password "secure_password" and password confirmation "secure_password"
+    Then response status code is 201
+    And response contains refreshToken "new-token" and the user with id "1235"
+
+  Scenario: Failed registration with used username, arbitrary password and password confirmation.
+    When user requests registration with username "user", password "pig-system" and password confirmation "pig-system"
+    Then response status code is 409
+    And response contains key "message" and value "User already exists"
+    And response contains key "status" and value "conflict"
+
+  Scenario: Failed registration with used username (other case), arbitrary password and password confirmation.
+    When user requests registration with username "User", password "pig-system" and password confirmation "pig-system"
+    Then response status code is 409
+    And response contains key "message" and value "User already exists"
+    And response contains key "status" and value "conflict"
+
+  Scenario: Failed registration with username, password and deviating password confirmation.
+    When user requests registration with username "kangaroo", password "pig-system" and password confirmation "i-love-capitalism"
```

Comment:
password confirmation could be checked by the FE already?

Reply:
To be honest, I would have done both, just for the sake of completeness